File: CleanDb.cs
using System;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;

class CleanDb {
    static void Main() {
        string path = @"d:\antigravity\db.json";
        string text = File.ReadAllText(path, Encoding.UTF8);

        // Stage Names
        text = text.Replace("Nhß║¡n th├┤ng tin", "Nhận thông tin");
        text = text.Replace("Lß║Ñy S─ÉT", "Lấy SĐT");
        text = text.Replace("Khai th├c th├┤ng tin", "Khai thác thông tin");
        text = text.Replace("B├o gi├", "Báo giá");
        text = text.Replace("Th╞░╞ng l╞░ß╗ng", "Thương lượng");
        text = text.Replace("Th├nh c├┤ng", "Thành công");
        text = text.Replace("Thß║Ñt bß║i", "Thất bại");

        // Specific lead names & notes
        text = Regex.Replace(text, @"Kh├ích Messenger Remote", "Khách Messenger Remote");
        text = Regex.Replace(text, @"Kh├ích Messenger 999", "Khách Messenger 999");
        text = Regex.Replace(text, @"Kh├ích Messenger", "Khách Messenger");
        text = Regex.Replace(text, @"D├║ng T├║c|D├║ng t├║c|D╞░╞íng T├│c", "Dương Tóc");
        text = Regex.Replace(text, @"Anh Ph╞░╞íng", "Anh Phương");
        text = Regex.Replace(text, @"Minh Nguyß╗àn", "Minh Nguyễn");
        text = Regex.Replace(text, @"Hu╞░╞íng Phß║ím|Hu╞░╞íng Phạ", "Huơng Phạm");
        text = Regex.Replace(text, @"Xu├ón H├ái ─É├¡nh|Xu├ón Hß║úi Đinh|Xuân Hải Dinh", "Xuân Hải Đinh");
        text = Regex.Replace(text, @"─É├¡nh Ph├║c An|Dinh Phúc An", "Đinh Phúc An");
        text = Regex.Replace(text, @"Ho├óng Th├╣y Du╞░╞íng", "Hoàng Thùy Dương");
        text = Regex.Replace(text, @"Phß║ím Thuß║¡n", "Phạm Thuận");
        text = Regex.Replace(text, @"Mai Hß╗Öng VPP", "Mai Hồng VPP");
        text = Regex.Replace(text, @"Ho├óng Ph├ít Koffmann", "Hoàng Phát Koffmann");
        text = Regex.Replace(text, @"V├▓ng bi Ph├║ Qu├╜", "Vòng bi Phú Quý");
        text = Regex.Replace(text, @"Nha Phuong B├╣i", "Nha Phuong Bùi");
        text = Regex.Replace(text, @"Quß╗æc Kh├ính", "Quốc Khánh");
        text = Regex.Replace(text, @"Minh T├ím", "Minh Tâm");
        text = Regex.Replace(text, @"Bß║úo Ngß╗ìc Rice", "Bảo Ngọc Rice");
        text = Regex.Replace(text, @"S╞í n Quang L├ím", "Sơn Quang Lâm");
        text = Regex.Replace(text, @"Phß║ím Thß╗ï Anh Ngß╗ìc", "Phạm Thị Anh Ngọc");
        text = Regex.Replace(text, @"Ho├óng C╞░╞íng Biz", "Hoàng Cường Biz");
        text = Regex.Replace(text, @"V┼⌐ Ngß╗ìc Huyß╗ün", "Vũ Ngọc Huyền");
        text = Regex.Replace(text, @"Trß║ºn Hiß║┐u", "Trần Hiếu");
        text = Regex.Replace(text, @"H╞░╞íng V┼⌐", "Hương Vũ");
        text = Regex.Replace(text, @"Ruby Nguyß╗ün", "Ruby Nguyễn");
        text = Regex.Replace(text, @"Diß╗åm Quß╗│nh|Điß╗âm Quß╗│nh", "Điểm Quỳnh");

        // Generic character mapping
        text = text.Replace("─É", "Đ").Replace("─æ", "đ");
        text = text.Replace("├║", "ú").Replace("├í", "á").Replace("├¡", "í").Replace("├┤", "ô");
        text = text.Replace("├¬", "ê").Replace("├á", "à").Replace("├¿", "è").Replace("├╣", "ù").Replace("├╜", "ý");
        text = text.Replace("ß╗a", "ẩ").Replace("ß╗å", "ổ").Replace("ß╗à", "ề").Replace("ß╗ï", "ị");
        text = text.Replace("ß╗ì", "ỉ").Replace("ß╗Å", "ỏ").Replace("ß╗ü", "ụ").Replace("ß╗ñ", "ủ");
        text = text.Replace("ß╗ª", "ữ").Replace("ß╗¿", "ừ").Replace("ß╗«", "ứ").Replace("ß╗░", "ử").Replace("ß╗▓", "ữ");
        text = text.Replace("ß║ím", "ạm").Replace("ß║í", "ạ").Replace("ß║º", "ầ").Replace("ß║┐", "ế");
        text = text.Replace("ß║╜", "ẽ").Replace("ß║╖", "ặt").Replace("ß╗ö", "ố");
        text = text.Replace("ß║╢", "ắ").Replace("ß║¡", "ận").Replace("ß║╟", "ẵ").Replace("ß║«", "ẳ");
        text = text.Replace("ß║▒", "ằ").Replace("ß║¯", "ắ").Replace("ß║£", "ả");
        text = text.Replace("ß║¥", "ấ").Replace("ß║§", "ẩ").Replace("ß║¨", "ẫ");
        text = text.Replace("ß║©", "ậ").Replace("ß║ª", "ẽ").Replace("ß║®", "ẽ");
        text = text.Replace("ß║°", "ề").Replace("ß║±", "ể").Replace("ß║²", "ễ");
        text = text.Replace("ß║³", "ệ").Replace("ß║´", "ỉ").Replace("ß║¶", "ọ");
        text = text.Replace("ß║·", "ỏ").Replace("ß║¸", "ố").Replace("ß║º", "ổ");
        text = text.Replace("ß║»", "ỗ").Replace("ß║¼", "ộ").Replace("ß║½", "ớ").Replace("ß║¾", "ờ");
        text = text.Replace("ß║¿", "ở").Replace("b─âng", "băng").Replace("l╞░ß╗¢i", "lưới");
        text = text.Replace("th╞░ß╗¥ng", "thường").Replace("k├¡ch", "kích").Replace("th╞░ß╗¢c", "thước");
        text = text.Replace("bß║n", "bản").Replace("rß╗Öng", "rộng").Replace("nß║╖ng", "nặng");

        File.WriteAllText(path, text, new UTF8Encoding(false));
        Console.WriteLine("CleanDb executed successfully!");
    }
}

File: clean_db.py
import json, re

def clean_text(s):
    if not isinstance(s, str):
        return s
    
    # Decoders for specific lead names & notes
    s = re.sub(r'Kh├ích Messenger Remote', 'Khách Messenger Remote', s)
    s = re.sub(r'Kh├ích Messenger 999', 'Khách Messenger 999', s)
    s = re.sub(r'Kh├ích Messenger', 'Khách Messenger', s)
    s = re.sub(r'D├║ng T├║c|D├║ng t├║c|D╞░╞íng T├│c', 'Dương Tóc', s)
    s = re.sub(r'Anh Ph╞░╞íng', 'Anh Phương', s)
    s = re.sub(r'Minh Nguyß╗àn', 'Minh Nguyễn', s)
    s = re.sub(r'Hu╞░╞íng Phß║ím|Hu╞░╞íng Phạ', 'Huơng Phạm', s)
    s = re.sub(r'Xu├ón H├ái ─É├¡nh|Xu├ón Hß║úi Đinh|Xuân Hải Dinh', 'Xuân Hải Đinh', s)
    s = re.sub(r'─É├¡nh Ph├║c An|Dinh Phúc An', 'Đinh Phúc An', s)
    s = re.sub(r'Ho├óng Th├╣y Du╞░╞íng', 'Hoàng Thùy Dương', s)
    s = re.sub(r'Phß║ím Thuß║¡n', 'Phạm Thuận', s)
    s = re.sub(r'Mai Hß╗Öng VPP', 'Mai Hồng VPP', s)
    s = re.sub(r'Ho├óng Ph├ít Koffmann', 'Hoàng Phát Koffmann', s)
    s = re.sub(r'V├▓ng bi Ph├║ Qu├╜', 'Vòng bi Phú Quý', s)
    s = re.sub(r'Nha Phuong B├╣i', 'Nha Phuong Bùi', s)
    s = re.sub(r'Quß╗æc Kh├ính', 'Quốc Khánh', s)
    s = re.sub(r'Minh T├ím', 'Minh Tâm', s)
    s = re.sub(r'Bß║úo Ngß╗ìc Rice', 'Bảo Ngọc Rice', s)
    s = re.sub(r'S╞í n Quang L├ím', 'Sơn Quang Lâm', s)
    s = re.sub(r'Phß║ím Thß╗ï Anh Ngß╗ìc', 'Phạm Thị Anh Ngọc', s)
    s = re.sub(r'Ho├óng C╞░╞íng Biz', 'Hoàng Cường Biz', s)
    s = re.sub(r'V┼⌐ Ngß╗ìc Huyß╗ün', 'Vũ Ngọc Huyền', s)
    s = re.sub(r'Trß║ºn Hiß║┐u', 'Trần Hiếu', s)
    s = re.sub(r'H╞░╞íng V┼⌐', 'Hương Vũ', s)
    s = re.sub(r'Ruby Nguyß╗ün', 'Ruby Nguyễn', s)
    s = re.sub(r'Diß╗åm Quß╗│nh|Điß╗âm Quß╗│nh', 'Điểm Quỳnh', s)

    # Stage names
    s = s.replace('Nhß║¡n th├┤ng tin', 'Nhận thông tin')
    s = s.replace('Lß║Ñy S─ÉT', 'Lấy SĐT')
    s = s.replace('Khai th├c th├┤ng tin', 'Khai thác thông tin')
    s = s.replace('B├o gi├', 'Báo giá')
    s = s.replace('Th╞░╞ng l╞░ß╗ng', 'Thương lượng')
    s = s.replace('Th├nh c├┤ng', 'Thành công')
    s = s.replace('Thß║Ñt bß║i', 'Thất bại')

    # Character-level CP437 mapping
    replacements = [
        ('─É', 'Đ'), ('─æ', 'đ'), ('├║', 'ú'), ('├í', 'á'), ('├¡', 'í'), ('├┤', 'ô'),
        ('├¬', 'ê'), ('├á', 'à'), ('├¿', 'è'), ('├╣', 'ù'), ('├╜', 'ý'),
        ('ß╗a', 'ẩ'), ('ß╗å', 'ổ'), ('ß╗à', 'ề'), ('ß╗ï', 'ị'), ('ß╗ì', 'ỉ'),
        ('ß╗Å', 'ỏ'), ('ß╗ü', 'ụ'), ('ß╗ñ', 'ủ'), ('ß╗ª', 'ữ'), ('ß╗¿', 'ừ'),
        ('ß╗«', 'ứ'), ('ß╗░', 'ử'), ('ß╗▓', 'ữ'), ('ß║ím', 'ạm'), ('ß║í', 'ạ'),
        ('ß║º', 'ầ'), ('ß║┐', 'ế'), ('ß║╜', 'ẽ'), ('ß║╖', 'ặt'), ('ß╗ö', 'ố'),
        ('ß║╢', 'ắ'), ('ß║¡', 'ận'), ('ß║╟', 'ẵ'), ('ß║«', 'ẳ'), ('ß║▒', 'ằ'),
        ('ß║¯', 'ắ'), ('ß║£', 'ả'), ('ß║¥', 'ấ'), ('ß║§', 'ẩ'), ('ß║¨', 'ẫ'),
        ('ß║©', 'ậ'), ('ß║ª', 'ẽ'), ('ß║®', 'ẽ'), ('ß║°', 'ề'), ('ß║±', 'ể'),
        ('ß║²', 'ễ'), ('ß║³', 'ệ'), ('ß║´', 'ỉ'), ('ß║¶', 'ọ'), ('ß║·', 'ỏ'),
        ('ß║¸', 'ố'), ('ß║º', 'ổ'), ('ß║»', 'ỗ'), ('ß║¼', 'ộ'), ('ß║½', 'ớ'),
        ('ß║¾', 'ờ'), ('ß║¿', 'ở'), ('b─âng', 'băng'), ('l╞░ß╗¢i', 'lưới'),
        ('th╞░ß╗¥ng', 'thường'), ('k├¡ch', 'kích'), ('th╞░ß╗¢c', 'thước'),
        ('bß║n', 'bản'), ('rß╗Öng', 'rộng'), ('nß║╖ng', 'nặng')
    ]
    for old, new in replacements:
        s = s.replace(old, new)

    return s

def clean_obj(obj):
    if isinstance(obj, dict):
        return {k: clean_obj(v) for k, v in obj.items()}
    elif isinstance(obj, list):
        return [clean_obj(item) for item in obj]
    elif isinstance(obj, str):
        return clean_text(obj)
    else:
        return obj

with open('db.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

cleaned_data = clean_obj(data)

with open('db.json', 'w', encoding='utf-8') as f:
    json.dump(cleaned_data, f, ensure_ascii=False, indent=2)

print("Python clean_db.py executed successfully!")
